src/librustc/traits/error_reporting.rs

```diff
@@ -551,6 +551,24 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
                         let OnUnimplementedNote { message, label }
                             = self.on_unimplemented_note(trait_ref, obligation);
                         let have_alt_message = message.is_some() || label.is_some();
+                        let span = match self.tcx.sess.codemap().span_to_snippet(span) {
+                            Ok(ref _s) if _s.starts_with("for ") => {
```

Comment:
style nit: `Ok(ref s) if s.starts_with("for ")` ([pretty sure](https://play.rust-lang.org/?gist=3fd8dec472b33b02f0b34e92e1bdec15&version=stable) the unused-variables lint is in fact smart enough to count the `if` guard as a use)

Reply:
For some reason I was convinced that there was a lint that complained at patterns that where only used in match guards... Must have been a different case.

Reply:
You might be thinking of how the dead-code lint complains about enum variants that are never constructed, even if they're matched on?